src/tool/hpcrun/audit/fake-auditor.c

```diff
@@ -275,15 +269,27 @@ int update_shadow_dl(struct dl_phdr_info* map, size_t sz, void* args_vp) {
         end = map->dlpi_phdr[i].p_vaddr + map->dlpi_phdr[i].p_memsz;
     }
   }
+
   entry->entry.start = (void*)map->dlpi_addr + start;
   entry->entry.end = (void*)map->dlpi_addr + end;
 
+  entry->entry.path = NULL;
+  if(args->index == 0 && map->dlpi_name[0] == '\0')
+    entry->entry.path = realpath((const char*)getauxval(AT_EXECFN), NULL);
+  else if (entry->entry.start == getauxval(AT_SYSINFO_EHDR))
+    entry->entry.path = realpath(vdso_path, NULL);
```

Comment:
The vDSO condition should still be stronger (more certain) than the main executable condition, so these two should be swapped.

Reply:
done